### PR TITLE
feat(cvsb-19580): Enrich error events with more essential context

### PR DESCRIPTION
**Pull request by FBarrassDVSA**
Enrich error events with more essential context
Added ATF and Vehicle VIN to Sentry events
https://jira.dvsacloud.uk/browse/CVSB-19580

## Checklist
- [x] Code has been tested manually
- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/sentry-ionic-errorhandler.ts [hunk 1]
```diff
@@ -3,25 +3,53 @@ import { IonicErrorHandler } from 'ionic-angular';
...
+import { VehicleModel } from '../models/vehicle/vehicle.model';
+import { TestModel } from '../models/tests/test.model';
 
 @Injectable()
 export class SentryIonicErrorHandler extends IonicErrorHandler {
-  constructor(private vaultService: VaultService) {
+  visit: VisitModel;
+  test: TestModel;
+  vehicle: VehicleModel;
+  constructor(private vaultService: VaultService, public visitService: VisitService) {
```
Comment (alleypa) [inline src/app/sentry-ionic-errorhandler.ts:16]: use `private visitService` instead

Comment (alleypa) [inline src/app/sentry-ionic-errorhandler.ts:16]: use private visitService instead

File: src/app/sentry-ionic-errorhandler.ts [hunk 2]
```diff
@@ -3,25 +3,53 @@ import { IonicErrorHandler } from 'ionic-angular';
...
+    const tests = this.visitService.visit.tests;
+
+    if (tests.length) {
+      this.test = tests.slice(-1)[0];
+      if (this.test.vehicles) {
+        this.vehicle = this.test.vehicles.slice(-1)[0];
+      }
+    }
+    return this.vehicle.vin && !this.test.status ? this.vehicle.vin : 'N/A';
+  }
```
Comment (alleypa) [inline src/app/sentry-ionic-errorhandler.ts:54]: Better delicate this responsibilites to the visit.service.ts

File: src/app/sentry-ionic-errorhandler.ts (hunk 1 above)
Comment (FBarrassDVSA) [reply src/app/sentry-ionic-errorhandler.ts:16]: updated this now to use private visitService.

File: src/app/sentry-ionic-errorhandler.ts (hunk 2 above)
Comment (FBarrassDVSA) [reply src/app/sentry-ionic-errorhandler.ts:54]: getting the current ATF and VIN is now done from visit.service.ts.

File: src/providers/visit/visit.service.ts [hunk 3]
```diff
@@ -103,6 +112,19 @@ export class VisitService {
...
 
+  getCurrentATF(): string {
+    return this.visit.testStationName ? this.visit.testStationName : 'N/A';
+  }
+
+  getCurrentVIN(): string {
+    const test = this.getLatestTest();
+    let vehicle;
+    if (test)
+      vehicle = this.getLatestVehicle();
```
Comment (alleypa) [inline src/providers/visit/visit.service.ts:123]: User `if (test) { }` with curly braces

Comment (FBarrassDVSA) [reply src/providers/visit/visit.service.ts:123]: Will update to add curly braces.

**Pull request closed without merging**